src/Microsoft.OData.Core/UriParser/Aggregation/AggregateExpression.cs

```diff
@@ -6,83 +6,39 @@
 
 namespace Microsoft.OData.UriParser.Aggregation
 {
-    using Microsoft.OData.Edm;
-    using Microsoft.OData.UriParser;
-
     /// <summary>
-    /// A aggregate expression representing a aggregation transformation.
+    /// Enumeration of the possible types of aggregations.
     /// </summary>
-    public sealed class AggregateExpression
+    public enum AggregateExpressionType
     {
-        private readonly AggregationMethod method;
-
-        private readonly SingleValueNode expression;
-
-        private readonly string alias;
-
-        private readonly IEdmTypeReference typeReference;
+        /// <summary>Value used to treat non recognized aggregations.</summary>
+        None = 0,
 
-        /// <summary>
-        /// Create a AggregateExpression.
-        /// </summary>
-        /// <param name="expression">The aggregation expression.</param>
-        /// <param name="method">The <see cref="AggregationMethod"/>.</param>
-        /// <param name="alias">The aggregation alias.</param>
-        /// <param name="typeReference">The <see cref="IEdmTypeReference"/> of this aggregate expression.</param>
-        public AggregateExpression(SingleValueNode expression, AggregationMethod method, string alias, IEdmTypeReference typeReference)
-        {
-            ExceptionUtils.CheckArgumentNotNull(expression, "expression");
-            ExceptionUtils.CheckArgumentNotNull(alias, "alias");
-            ExceptionUtils.CheckArgumentNotNull(typeReference, "typeReference");
-
-            this.expression = expression;
-            this.method = method;
-            this.alias = alias;
-            this.typeReference = typeReference;
-        }
+        /// <summary>Aggregation of a single value property.</summary>
+        PropertyAggregate = 1,
 
-        /// <summary>
-        /// Gets the aggregation expression.
-        /// </summary>
-        public SingleValueNode Expression
-        {
-            get
-            {
-                return expression;
-            }
-        }
+        /// <summary>Aggregation of a entity set property.</summary>
+        EntitySetAggregate = 2
+    }
 
-        /// <summary>
-        /// Gets the <see cref="AggregationMethod"/>.
-        /// </summary>
-        public AggregationMethod Method
+    /// <summary>
+    /// A aggregate expression representing a aggregation transformation.
+    /// </summary>
+    public abstract class AggregateExpression

```

Comment:
Done it. Made some little changes in the naming conventions to make the classes functions a little bit clearer. Could you please check if it is OK?